src/test/java/htsjdk/samtools/cram/structure/SliceTests.java

```diff
@@ -72,238 +75,180 @@ public void testFailsMD5Check() throws IOException {
-        Assert.assertEquals(slice.alignmentStart, Slice.NO_ALIGNMENT_START);
-        Assert.assertEquals(slice.alignmentSpan, Slice.NO_ALIGNMENT_SPAN);
+        return record;
     }
 
-
-    @Test
-    public void testSingleAndUnmappedBuild() {
+    private List<CramCompressionRecord> getSingleRefRecords() {
         final List<CramCompressionRecord> records = new ArrayList<>();
+        for (int i = 0; i < TEST_RECORD_COUNT; i++) {
+            records.add(createRecord(i, 0));
+        }
+        return records;
+    }
 
-        final CramCompressionRecord single = new CramCompressionRecord();
-        single.readBases = "AAA".getBytes();
-        single.qualityScores = "!!!".getBytes();
-        single.readLength = 3;
-        single.readName = "1";
-        single.sequenceId = 0;
-        single.alignmentStart = 1;
-        single.setLastSegment(true);
-        single.readFeatures = Collections.emptyList();
-        single.setSegmentUnmapped(false);
-
-        records.add(single);
-
-        final CramCompressionRecord unmapped = new CramCompressionRecord();
-        unmapped.readBases = "AAA".getBytes();
-        unmapped.qualityScores = "!!!".getBytes();
-        unmapped.readLength = 3;
-        unmapped.readName = "unmapped";
-        unmapped.sequenceId = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
-        unmapped.alignmentStart = SAMRecord.NO_ALIGNMENT_START;
-        unmapped.setLastSegment(true);
-        unmapped.readFeatures = Collections.emptyList();
-        unmapped.setSegmentUnmapped(true);
-
-        records.add(unmapped);
-
-        final CompressionHeader header = new CompressionHeaderFactory().build(records, null, true);
+    private List<CramCompressionRecord> getMultiRefRecords() {
+        final List<CramCompressionRecord> records = new ArrayList<>();
+        for (int i = 0; i < TEST_RECORD_COUNT; i++) {
+            records.add(createRecord(i, i));
+        }
+        return records;
+    }
 
-        final Slice slice = Slice.buildSlice(records, header);
+    private List<CramCompressionRecord> getNoRefRecords() {
+        final List<CramCompressionRecord> records = new ArrayList<>();
+        for (int i = 0; i < TEST_RECORD_COUNT; i++) {
+            records.add(createRecord(i, SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX));
+        }
+        return records;
+    }
 
-        Assert.assertEquals(slice.nofRecords, 2);
-        Assert.assertEquals(slice.sequenceId, Slice.MULTI_REFERENCE);
-        Assert.assertEquals(slice.alignmentStart, Slice.NO_ALIGNMENT_START);
-        Assert.assertEquals(slice.alignmentSpan, Slice.NO_ALIGNMENT_SPAN);
+    private List<CramCompressionRecord> getNoStartRecords() {
+        final List<CramCompressionRecord> records = new ArrayList<>();
+        for (int i = 0; i < TEST_RECORD_COUNT; i++) {
+            final CramCompressionRecord rec = createRecord(i, i);
+            rec.alignmentStart = SAMRecord.NO_ALIGNMENT_START;
```

Comment:
Same comment here. Some of these records are mapped/unplaced (though for a different reason than the one above, since these have a seq index but no start).